9 user authetication

OscarFMdev opened the pull request:
# PR description 📄 

In this branch @Mhamad-Raad  and I:

- Added devise to the project 👤 
- Updated views 🪟

### Thank you for reviewing our code! 🙌🏽

OscarFMdev committed:
Add devise to user

Mhamad-Raad committed:
add logout function

danmainah commented on app/models/user.rb:
```diff
@@ -1,8 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
- [ ] @OscarFMdev and @Mhamad-Raad kindly ensure your app asks for confirmation of email when signing up, kindly add the device confirmable in your user model, this is a project requirement
![image](https://user-images.githubusercontent.com/61408860/216149742-785d71e4-6ccf-4b81-8cc5-80765591d9d1.png)

Whoistolu replied:
- Well resolved @OscarFMdev

danmainah commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -0,0 +1,25 @@
+<h2>Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "current-password" %>
+  </div>
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+<%= render "devise/shared/links" %>
```
- [ ] @team kindly ensure your login/registration pages match the wireframe design given below is a comparison for  better understanding :


Design Given          |  Your design     
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/61408860/216150334-c518966d-112c-4fd8-9394-d368c1ee770a.png) |  ![image](https://user-images.githubusercontent.com/61408860/216150402-58ad092e-19e2-4810-9970-8dfafa075df1.png)

OscarFMdev replied:
Hello @danmainah, for this step should we remove the sign-up, remember me and forgot your password? or we should only add a username instead of an email? Thank you! 🙌🏽

Whoistolu replied:
- Well resolved @OscarFMdev

danmainah reviewed (changes requested):
Hi @OscarFMdev and @Mhamad-Raad , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [optional] It would be nice if you added details of the second author in the readme file 
![image](https://user-images.githubusercontent.com/61408860/216150861-89c55a9c-3159-469a-bf1e-0192aa302512.png)


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

OscarFMdev committed:
Update README.md

OscarFMdev committed:
Add confirmable feature

OscarFMdev committed:
Fix linter errors 🤖

OscarFMdev committed:
Add styles to login form

OscarFMdev commented:
# Update after first code review
### Dear code reviewer 
These are the issues that I've fixed from the previous review:
1. Added a confirmation feature:
<img width="447" alt="Screenshot 2023-02-01 at 22 08 17" src="https://user-images.githubusercontent.com/79666590/216377001-a7b8dade-5f7e-4b78-99df-6cccbfa46b8a.png">

2. Updated view to match the wireframe (I still have the links below because I consider these links helpful for the user)
<img width="456" alt="Screenshot 2023-02-02 at 10 02 32" src="https://user-images.githubusercontent.com/79666590/216377464-f1ce6e74-5d98-480b-b94b-9f7f61250c5c.png">

### Thank you for taking the time to review my code! 🙌🏽

Whoistolu reviewed (approved):
### Hi @OscarFMdev 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

- [ ] [OPTIONAL] Kindly update the license year to 2023 since this I the year you built this project.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

OscarFMdev commented:
Thank you for the optional suggestion @Whoistolu , I'll update it on the next  branch 🙌🏽
